hooks/tk-multi-publish2/ingest/ingest_files.py

```diff
@@ -0,0 +1,198 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import copy
+import glob
+import pprint
+import traceback
+
+import sgtk
+from sgtk import TankError
+
+from tank.util.shotgun import get_sg_connection
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class IngestFilesPlugin(HookBaseClass):
+    """
+    Inherits from PublishFilesPlugin
+    """
+
+    @property
+    def description(self):
+        """
+        Verbose, multi-line description of what the plugin does. This can
+        contain simple html for formatting.
+        """
+
+        desc = super(IngestFilesPlugin, self).description
+
+        return desc + "<br><br>" + """
+        After Ingesting the file a publishedFile gets created. The publishedFile will also be linked to a Plate entity.
+        """
+
+    def publish(self, task_settings, item):
+        """
+        Executes the publish logic for the given item and task_settings.
+
+        :param task_settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the task_settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        """
+
+        super(IngestFilesPlugin, self).publish(task_settings, item)
+
+        # create a plate entity after the publish has went through successfully.
```

Comment:
"has gone through". Also, you should probably test that the parent process has succeeded before continuing on.

Reply:
you want me to check for "sg_publish_data" on the item? Or is there any better way to know if the process succeeded?

Reply:
That's the best way.